internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -445,24 +453,35 @@ func (c *enrollCmd) prepareFleetTLS() error {
 
 func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	err := c.daemonReload(ctx)
+	if err != nil &&
+		(errors.Is(err, context.DeadlineExceeded) ||
+			errors.Is(err, context.Canceled)) {
+		return fmt.Errorf("could not reload daemon: %w", err)
+	}
 	if err == nil {
 		return nil
 	}
```

Comment:
Maybe  we can simplify a bit
```suggestion
	if err == nil {
		return nil
	}
	if errors.Is(err, context.DeadlineExceeded) ||
			errors.Is(err, context.Canceled) {
		return fmt.Errorf("could not reload daemon: %w", err)
	}
```